### ISSUE TITLE
pathless layouts issue with v2 routing on `1.17.1`

### ISSUE THREAD

**Issue by theosenoussaoui** [remix-run/remix#6645]
### What version of Remix are you using?

1.17.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

- Either fork this [reproduction repository](https://github.com/theosenoussaoui/repro-remix) or go on this [Codesandbox](https://codesandbox.io/p/github/theosenoussaoui/repro-remix).
- Go on the `/test/catalog` route.

### Expected Behavior

I should be able to to go on the `/test/catalog` route without having a 404 error.

### Actual Behavior

I'm having a 404 error on the `/test/catalog` route.

**Comment by theosenoussaoui** [comment]
For additional information, here is the output of `npx remix routes` : 

```
npx remix routes
<Routes>
  <Route file="root.tsx">
    <Route path="test" file="routes/test.tsx">
      <Route path="test" file="routes/test._pathless.tsx">
        <Route path="catalog" file="routes/test._pathless.catalog.tsx" />
      </Route>
      <Route index file="routes/test._index.tsx" />
    </Route>
  </Route>
</Routes>
```

I tried asking on the Remix for help and I've been advised to create an issue here, please do not hesitate to tell me if issuing routing issues for remix v2 isn't allowed yet.
I'm not sure if you guys are opened for contributions on this, but I'd love to try (at least) to find the issue for this.

**Comment by kiliman** [comment]
This looks like a bug in `v2` routing. Notice how the entry for the _pathless_ route still has a `path` attribute. If you go to `/test/test/catalog` you'll see the route.
```html
<Routes>
  <Route file="root.tsx">
    <Route path="test" file="routes/test.tsx">
      <!-- This is a pathless route, so path attribute should not be here -->
      <Route path="test" file="routes/test._pathless.tsx">
        <Route path="catalog" file="routes/test._pathless.catalog.tsx" />
      </Route>
      <Route index file="routes/test._index.tsx" />
    </Route>
  </Route>
</Routes>
```

With that being said, `remix-flat-routes` works correctly. Add the package and update your *remix.config.js* file:
```js
/** @type {import('@remix-run/dev').AppConfig} */

const { flatRoutes } = require("remix-flat-routes");

module.exports = {
  ignoredRouteFiles: ["**/*"],
  routes: async (defineRoutes) => {
    return flatRoutes("routes", defineRoutes, { enableUniqueIdCheck: false });
  },
  serverModuleFormat: "cjs",
  future: {
    v2_errorBoundary: true,
    v2_headers: true,
    v2_meta: true,
    v2_normalizeFormMethod: true,
    v2_routeConvention: true,
  },
};
```
Afterward, `npx remix routes` looks like this:
```html
<Routes>
  <Route file="root.tsx">
    <Route path="test" file="routes/test.tsx">
      <Route file="routes/test._pathless.tsx">
        <Route path="catalog" file="routes/test._pathless.catalog.tsx" />
      </Route>
      <Route index file="routes/test._index.tsx" />
    </Route>
  </Route>
</Routes>
```

**Comment by theosenoussaoui** [comment]
> This looks like a bug in `v2` routing. Notice how the entry for the _pathless_ route still has a `path` attribute. If you go to `/test/test/catalog` you'll see the route.
> 
> ```
> <Routes>
>   <Route file="root.tsx">
>     <Route path="test" file="routes/test.tsx">
>       
>       <Route path="test" file="routes/test._pathless.tsx">
>         <Route path="catalog" file="routes/test._pathless.catalog.tsx" />
>       </Route>
>       <Route index file="routes/test._index.tsx" />
>     </Route>
>   </Route>
> </Routes>
> ```
> 
> With that being said, `remix-flat-routes` works correctly. Add the package and update your _remix.config.js_ file:
> 
> ```js
> /** @type {import('@remix-run/dev').AppConfig} */
> 
> const { flatRoutes } = require("remix-flat-routes");
> 
> module.exports = {
>   ignoredRouteFiles: ["**/*"],
>   routes: async (defineRoutes) => {
>     return flatRoutes("routes", defineRoutes, { enableUniqueIdCheck: false });
>   },
>   serverModuleFormat: "cjs",
>   future: {
>     v2_errorBoundary: true,
>     v2_headers: true,
>     v2_meta: true,
>     v2_normalizeFormMethod: true,
>     v2_routeConvention: true,
>   },
> };
> ```
> 
> Afterward, `npx remix routes` looks like this:
> 
> ```
> <Routes>
>   <Route file="root.tsx">
>     <Route path="test" file="routes/test.tsx">
>       <Route file="routes/test._pathless.tsx">
>         <Route path="catalog" file="routes/test._pathless.catalog.tsx" />
>       </Route>
>       <Route index file="routes/test._index.tsx" />
>     </Route>
>   </Route>
> </Routes>
> ```

Sure, I'll try to adapt this to our project. However, I just need to know if the `remix-flat-routes` will have breaking changes compared to `remix` own `v2` routing. Eventually, I guess the goal is to remove this package to have as less dependencies as possible.

Either way, thanks for the answer, I'll try to implement this solution on our project as soon as I can and report back !

**Comment by kiliman** [comment]
`remix-flat-routes` was the initial version of the flat routes (v2) specification. They ported it as a core feature. My package is a superset (it has a lot more functionality than v2). It will always be compatible with the core version.

**Comment by brophdawg11** [comment]
This is fixed by #6649 and should be available in `1.18.0` once released
